src/Overcrowdin/UpdateCommand.cs

```diff
@@ -23,18 +23,15 @@ public class Options : GlobalOptions, IFileOptions
 			// TODO: Add option for update approval -- default to Update_as_unapproved
 		}
 
-		public static async Task<int> UpdateFilesInCrowdin(IConfiguration config, Options opts, IFileSystem fileSystem)
+		public static async Task<int> UpdateFilesInCrowdin(IConfiguration config, Options opts, IFileSystem fileSystem, ICrowdinClientFactory apiFactory)
 		{
-			var crowdin = CrowdinCommand.GetClient();
-
-			var projectId = config["project_identifier"];
-			var credentials = CommandUtilities.GetCredentialsFromConfiguration(config);
+			var credentials = await CommandUtilities.GetProjectSettingsFromConfiguration(config, opts.Branch, apiFactory);
 			if (credentials == null)
 			{
 				return 1;
 			}
-
-			var updateFileParametersList = new List<UpdateFileParameters>();
+			var uploadHelper = await CrowdInUploadHelper.Create(credentials, fileSystem, apiFactory);
+			var updateFileParametersList = new List<FileParameters>();
 			CommandUtilities.GetFileList(config, opts, fileSystem, updateFileParametersList, new SortedSet<string>());
 
 			if (!updateFileParametersList.Any())
```

Comment:
I would move this near the top of this method. No need to fetch credentials and make an upload helper if there's no files to add.